ui/uihandler.go

```diff
@@ -968,11 +972,14 @@ func RedirectToHTTPS(port string) {
 
 // setHostAndPort takes hostport in the form of hostname:port, ip:port or :port,
 // sets the urlHostPort variable and returns hostport unmodified.
-func setHostAndPort(hostport string) string {
-	if strings.HasPrefix(hostport, ":") {
-		urlHostPort = "localhost" + hostport
-		return hostport
+func setHostAndPort(inputPort string, addr *net.TCPAddr) {
```

Comment:
can add a test for this.

Reply:
the whole thing is quite yucky and has no unit tests... so far